scripts/nlp/pptext.py

```diff
@@ -1,8 +1,140 @@
+import spacy
+from spacy import displacy
+
 # preprocessing function
 
-def clean_data(df):
-    pass
+def clean_data(dataframe_column):
+    print(f'Started cleaning: the dataframe column from all the special characters\n')
+    dataframe_column = dataframe_column.str.replace('\xa0',' ')
+    dataframe_column = dataframe_column.str.replace('\|\|-\|\|','')
+
+    spec_chars = ["!",'"',"#","%","&","'","(",")",
+                  "*","+",",",".","/",";","<",
+                  "=",">","?","@","[","\\","]","^","_",
+                  "`","{","}","~","–"]
+    for char in spec_chars:
+        dataframe_column = dataframe_column.str.replace(char, ' ')
+
+    dataframe_column = dataframe_column.str.replace(r'^\s+|\s+$', '')
+    dataframe_column = dataframe_column.str.replace('    ',' ')
+    dataframe_column = dataframe_column.str.replace('  ',' ')
+    dataframe_column = dataframe_column.str.replace('  ',' ')
+    print(dataframe_column[13])
+
+    dataframe_column_clean = dataframe_column
+
+    all_corpus_text = []
+    for phrases in dataframe_column:
+        phrases = phrases.split(" ")
+        for words in phrases:
+            all_corpus_text.append(words)
+    return all_corpus_text , dataframe_column_clean
+
+
+# pass it a dataframe column and it will go over them to extract all the entities that it has and displays them
+def detect_entities_in_dataframe(dataframe_column, model_path):
+    for sentence in dataframe_column:
+        ## print named entities in phrases
+        nlp = spacy.load(model_path)
```

Comment:
de ce faci load la model pt fiecare exemplu? cand poti face load inainte de loop? si in loop doar sa prezici

Reply:
a fost mutat in afara functiei